src/main/java/org/mule/module/xml/internal/operation/XsltOperation.java

```diff
@@ -83,11 +82,6 @@ public String xsltTransform(@Content(primary = true) InputStream content,
     return withTransformer(xslt, transformer -> {
       bindParameters(transformer, contextProperties);
 
-      // TODO: Question: Should I put this in try-finally block to deal with the case something fails in between,
-      // and the transformer is left with the custom resolver?
-
-      // Save controller's OutputUriResolver to deal with transformer pooling
-      OutputURIResolver previousUriResolver = transformer.getUnderlyingController().getOutputURIResolver();
       // Set URI scheme correction resolver
       transformer.getUnderlyingController().setOutputURIResolver(new FileSchemeCorrectionOutputUriResolver());
```

Comment:
shouldn't this actually happen at the `create()` of `org.mule.module.xml.internal.operation.XsltOperation#createPooledObjectFactory` ?

Reply:
Completely agree.